src/server/routes/api/index.js

```diff
@@ -18,36 +17,76 @@ function getSqsPayload(object) {
   };
 }
 
+function isSqsDataValid(params, type) {
+  const { barcodes, protectCGD, action, email } = params;
+  if (type === 'transfer') {
+    const { bibRecordNumber } = params;
 
-export function updateMetadata(req, res, next) {
-  const params = req.body;
+    if (isEmpty(barcodes) || isEmpty(bibRecordNumber) || typeof protectCGD !== 'boolean' || isEmpty(action) || isEmpty(email)) {
+      return false;
+    }
+  }
 
-  if (!params) {
-    return res.status(400).json({
-      error: 'Error: malformed request; verify payload properties',
-      status: 400
-    });
+  if (type === 'update') {
+    if (isEmpty(barcodes) || typeof protectCGD !== 'boolean' || isEmpty(action) || isEmpty(email)) {
+      return false;
+    }
   }
 
-  const paramsAsJsonString = JSON.stringify(params);
-  const { api , region } = config.sqs;
-  aws.config.update({ region: region });
-  // Send to SQS
-  const sqsPayload = getSqsPayload({ message: paramsAsJsonString, queueUrl: api });
-  const sqs = new aws.SQS({ apiVersion: '2012-11-05' });
+  return true;
+}
+
+export function handleSqsDataProcessing(sqsClient, type) {
+  return (req, res, next) => {
+    const { api } = config.sqs;
+    const params = req.body;
+
+    if (!sqsClient) {
+      return res.status(503).json({
+        error: 'SQS Client is not defined',
+        status: 503
+      });
+    }
 
-  sqs.sendMessage(sqsPayload, (err, data) => {
-    if (err) {
-      console.log("Error", err);
+    if (!type || type === '') {
       return res.status(503).json({
-        error: err,
+        error: 'SQS transaction type is undefined',
```

Comment:
If the 'type' is invalid, it suggests our server is unable to to assign a proper `action` to it (since neither the user or AWS SQS is involved to put in the `action`), so I think the error code should be `500`

Reply:
I spoke about it with @thisisstephenbetts -- any 5xx range error is considered a server error. An internal server error makes sense as well, I don't mind changing it to that.

Reply:
I see. That's cool! It is just a suggestion. Final decision is of course up to you and @thisisstephenbetts!